app/src/main/java/ch/hikemate/app/ui/map/HikeDetailsScreen.kt

```diff
@@ -259,11 +380,20 @@ fun hikeDetailsMap(hike: DetailedHike): MapView {
           min(HikeDetailScreen.MAP_MAX_LONGITUDE, mapView.boundingBox.lonEast),
           HikeDetailScreen.MAP_BOUNDS_MARGIN)
     }
+    // This is useful for the first composition since there won't be any zoom or scroll events at
+    // first so the hike displays the possible facilities from the get go.
+    facilitiesViewModel.filterFacilitiesForDisplay(
+        facilities.value ?: emptyList(),
+        bounds = boundingBoxState.value,
+        zoomLevel = zoomLevelState.value,
+        hikeRoute = hike,
+        onSuccess = { MapUtils.displayFacilities(it, mapView, context) },
+        onNoFacilitiesForState = { /*Nothing since nothing is displayed yet*/})
```

Comment:
Not really. Since this is an `addOnFirstLayoutListener` it'll called during the composable execution, so the listener itself will only trigger once.